src/CommitmentsV2/SFA.DAS.CommitmentsV2.UnitTests/Application/Commands/UpdateApprenticeshipStopDateCommandHandlerTests.cs

```diff
@@ -0,0 +1,387 @@
+                {
+                    third.PropertyName.Should().Be("UserInfo");
+                    third.ErrorMessage.Should().Be("The User Info supplied must not be null and contain a UserId");
+                },
+                fourth =>
+                {
+                    fourth.PropertyName.Should().Be("StopDate");
+                    fourth.ErrorMessage.Should().Be("The StopDate must be supplied");
+                });
+        }
+
+
+        [Test, MoqAutoData]
+        public async Task Handle_WhenHandlingCommand_WithInvalidCallingParty_ThenShouldThrowDomainException(UpdateApprenticeshipStopDateCommand command)
+        {
+            // Arrange
+            var apprenticeship = await SetupApprenticeship(Party.Provider);
+
+            // Act
+            var exception = Assert.ThrowsAsync<DomainException>(async () => await _handler.Handle(command, new CancellationToken()));
+
+            // Assert
+            exception.DomainErrors.Should().BeEquivalentTo(new { ErrorMessage = "UpdateApprenticeshipStopDate is restricted to Employers only - Provider is invalid" });
+        }
+
+        [Test]
+        public async Task Handle_WhenHandlingCommand_WithInvalidApprenticeshipForStop_PaymentStatusCompleted_ThenShouldThrowDomainException()
+        {
+            // Arrange
+            var apprenticeship = await SetupApprenticeship(paymentStatus: PaymentStatus.Completed);
+            var command = new UpdateApprenticeshipStopDateCommand(apprenticeship.Cohort.EmployerAccountId, apprenticeship.Id, DateTime.UtcNow, new UserInfo());
+
+            // Act
+            var exception = Assert.ThrowsAsync<DomainException>(async () => await _handler.Handle(command, new CancellationToken()));
+
+            // Assert
+            exception.DomainErrors.Should().BeEquivalentTo(new { PropertyName = "newStopDate", ErrorMessage = "Apprenticeship must be stopped in order to update stop date" });
+        }
+
+        [Test]
+        public async Task Handle_WhenHandlingCommand_WhenValidatingApprenticeship_WithStopDateInFuture_ThenShouldThrowDomainException()
+        {
+            // Arrange
+            var stopDate = DateTime.UtcNow.AddMonths(1);
+            var apprenticeship = await SetupApprenticeship(paymentStatus: PaymentStatus.Withdrawn);            
+            var command = new UpdateApprenticeshipStopDateCommand(apprenticeship.Cohort.EmployerAccountId, apprenticeship.Id, stopDate, new UserInfo());
+
+            // Act
+            var exception = Assert.ThrowsAsync<DomainException>(async () => await _handler.Handle(command, new CancellationToken()));
+
+            // Assert
+            exception.DomainErrors.Should().BeEquivalentTo(new { PropertyName = "newStopDate", ErrorMessage = $"Invalid Date of Change. Date cannot be in the future." });
+        }
+
+
+        [Test]
+        public async Task Handle_WhenHandlingCommand_WhenValidatingApprenticeship_WithStopDateNotEqualStartDate_ThenShouldThrowDomainException()
```

Comment:
Test name suggests that the error is stop date != start date but it is not the case. It is stop date < start date

Reply:
changed the test method name